cpp/src/aztec3/circuits/rollup/base/native_base_rollup_circuit.cpp

```diff
@@ -89,6 +89,82 @@ std::vector<NT::fr> calculate_contract_leaves(BaseRollupInputs baseRollupInputs)
     return contract_leaves;
 }
 
+template <size_t N>
+NT::fr iterate_through_tree_via_sibling_path(NT::fr leaf,
+                                             NT::uint32 leafIndex,
+                                             std::array<NT::fr, N> siblingPath,
+                                             size_t startAtDepth = 0)
+{
+    for (size_t i = startAtDepth; i < siblingPath.size(); i++) {
```

Comment:
Santiago will be giving us only the sibling path from the root to subtree root, so the start debt would in practice always be 0.